### PR TITLE
vimPlugins.lze: init at 0.1.1

### PR DESCRIPTION
**Pull request by BirdeeHub**
I made this, and I almost completely deleted it because I was fully on board with lz-n again. However, I have been using it and lz-n simultaneosly and trying them out.

I wish to publish [lze](https://github.com/BirdeeHub/lze) after all. I like it a lot.

It works completely differently from lz-n and has a different handler api, but has a similar plugin spec.

It has been rewritten significantly AGAIN since I last went to upload it.

It has a much more respectful readme this time around too, as now I have more legitimately good things to say about both options.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by PerchunPak — APPROVED** [review]
LGTM, though nixpkgs-review would be nice

CC @mrcjkb @teto

**Comment by mrcjkb** [comment]
Result of `nixpkgs-review pr 339387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>lua51Packages.lze</li>
    <li>luaPackages.lze (lua52Packages.lze)</li>
    <li>lua53Packages.lze</li>
    <li>lua54Packages.lze</li>
    <li>luajitPackages.lze</li>
    <li>vimPlugins.lze</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Review by mrcjkb — APPROVED** [review]
LGTM.

@BirdeeHub fyi, you can run nixpkgs-review with `nix run .#nixpkgs-review -- pr <pr-number> --post` (replace `<pr-number>`, in this case with `339387`).
I've just run it for x86_64-linux.

See also: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#tested-compilation-of-all-pkgs-that-depend-on-this-change-using-nixpkgs-review

**Comment by BirdeeHub** [comment]
Ah, yeah ty. Still new here. Kinda thought that other people were supposed to be the ones to do stuff to make sure it worked on THEIR hardware. Some stuff still to learn about contributing :)

**Comment by BirdeeHub** [comment]
I also forgot about the --post option XD

**Comment by BirdeeHub** [comment]
Result of `nixpkgs-review pr 339387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>lua51Packages.lze</li>
    <li>luaPackages.lze (lua52Packages.lze)</li>
    <li>lua53Packages.lze</li>
    <li>lua54Packages.lze</li>
    <li>luajitPackages.lze</li>
    <li>vimPlugins.lze</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Comment by PerchunPak** [comment]
> Ah, yeah ty. Still new here. Kinda thought that other people were supposed to be the ones to run it to make sure it worked on THEIR hardware. Some stuff still to learn about contributing :)

(Still new here as well) I suppose simple packages, that don't use any specific hardware (cuda, drivers, etc), don't need to be tested by external contributor as Nix is pure and if something works on one system, it will work everywhere

**Comment by BirdeeHub** [comment]
Yeah Id be pretty surprised if this didnt work anywhere that you could run nvim XD I dont even refer to any filepaths anywhere XD

File: pkgs/development/lua-modules/overrides.nix [hunk 1]
```diff
@@ -544,6 +544,17 @@ in
...
 
+  lze  = prev.lze.overrideAttrs(oa: {
+    doCheck = lua.luaversion == "5.1";
+    nativeCheckInputs = [ final.nlua final.busted ];
+    checkPhase = ''
+      runHook preCheck
+      export HOME=$(mktemp -d)
+      busted --lua=nlua
+      runHook postCheck
+      '';
```
Comment (GaetanLepage) [inline pkgs/development/lua-modules/overrides.nix:555]:
```suggestion
    '';
```
Here you go, this is my contribution to this work X)

**Comment by PerchunPak** [comment]
> I also forgot about the --post option XD My PAT doesnt have permissions to post :( Guess I have to change that

Btw, you don't have to use --post. For example, I am too lazy to configure my GitHub key, so I just do `cat report.md` in a shell created by nixpkgs-review

**Comment by BirdeeHub** [comment]
Thats fair. It was easy enough tho :)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339387` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>lua51Packages.lze</li>
    <li>lua52Packages.lze</li>
    <li>lua53Packages.lze</li>
    <li>lua54Packages.lze</li>
    <li>luajitPackages.lze</li>
    <li>vimPlugins.lze</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339387` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>lua51Packages.lze</li>
    <li>lua52Packages.lze</li>
    <li>lua53Packages.lze</li>
    <li>lua54Packages.lze</li>
    <li>luajitPackages.lze</li>
    <li>vimPlugins.lze</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>lua51Packages.lze</li>
    <li>lua52Packages.lze</li>
    <li>lua53Packages.lze</li>
    <li>lua54Packages.lze</li>
    <li>luajitPackages.lze</li>
    <li>vimPlugins.lze</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 339387` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>lua51Packages.lze</li>
    <li>lua52Packages.lze</li>
    <li>lua53Packages.lze</li>
    <li>lua54Packages.lze</li>
    <li>luajitPackages.lze</li>
    <li>vimPlugins.lze</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Pull request merged**